src/v/utils/fragmented_vector.h

```diff
@@ -179,6 +179,8 @@ class fragmented_vector {
 
     const T& front() const { return _frags.front().front(); }
     const T& back() const { return _frags.back().back(); }
+    T& front() { return _frags.front().front(); }
+    T& back() { return _frags.back().back(); }
```

Comment:
is the back() element of the last fragment (vector) always the last element? i guess it is after a call to shrink_to_fit but i'm not sure of all the different scenarios now that the fragmented_vector API has grown quite a lot.

Reply:
Yes we still maintain that invariant yes.

Your question made me realize there is a bug in the `pop_back_n` implementation related to this so I have fixed that and enhanced the tests to check the `back` invariant.